src/components/menu/popoverMenu.tsx

```diff
@@ -0,0 +1,107 @@
+import './popoverMenu.css'
+
+import MoreVertIcon from '@mui/icons-material/MoreVert'
+import { useMediaQuery } from '@mui/material'
+import Drawer from '@mui/material/Drawer'
+import IconButton from '@mui/material/IconButton'
+import ListItemIcon from '@mui/material/ListItemIcon'
+import ListItemText from '@mui/material/ListItemText'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import { useTheme } from '@mui/material/styles'
+import Typography from '@mui/material/Typography'
+import type { ReactNode } from 'react'
+import React, { useState } from 'react'
+
+export interface PopoverMenuItem {
+  label: string
+  onClick: (event?: React.MouseEvent<HTMLElement>) => void
+  divider?: boolean
+  icon?: ReactNode
+  endDecorator?: ReactNode
+}
+
+export interface PopoverMenuProps {
+  menuItems: PopoverMenuItem[]
+  /** Aria-label to describe the button for assistive technology. */
+  ariaLabel: string
+  /** Provide your own customized icon. */
+  icon?: ReactNode
+}
+
+export default function PopoverMenu(props: PopoverMenuProps) {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null)
+  const theme = useTheme()
+
+  const isOpen = Boolean(anchorEl)
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'))
+
+  const PopoverMenuElement = isMobile ? Drawer : Menu
+
+  let menuBorderStyles: object = {
+    borderRadius: `${theme.shape.borderRadius}px`,
+    border: `1px solid ${theme.palette.divider}`,
+  }
+
+  if (isMobile) {
+    menuBorderStyles = {
+      ...menuBorderStyles,
+      borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+      borderBottom: 0,
+    }
+  }
+
+  const menuItems = props.menuItems.map((menuItem) => {
+    function handleOnClick() {
+      menuItem.onClick()
+      setAnchorEl(null)
+    }
```

Comment:
The event must be propagated to user defined click function.
```suggestion
    function handleOnClick(event: React.MouseEvent<HTMLElement>) {
      menuItem.onClick(event)
      setIsOpen(false)
    }
```

Reply:
Please add a test case to check the event propagation